### PR TITLE
feat(v3): Restructure API docs for v3

### LINKED ISSUE #5397: v3 management API
This PR:
- Adds an API reference doc for the  HTTP management API openapi spec (https://github.com/influxdata/granite/blob/main/openapi.yaml) (Cloud Dedicated only)
- Adds Management API examples from https://raw.githubusercontent.com/influxdata/granite/aea169033af0d7f7d6124c1a3cab61c085038bd2/scripts/test_http_api_v0_endpoints.sh
- Doesn't fetch changes for any specs except for the Management spec.
- Adds summaries and fixes descriptions in spec overrides.
- Removes versions in spec overrides.
- Adds configuration for generating API reference docs, Hugo frontmatter, setting the default API (for the [product]/api/ URL)
- Improves Hugo frontmatter descriptions and reuses titles and summaries from specs.
- Restructures `/api-docs` to better allow for multiple APIs (and customizations) per product
- Adds `[product]/api/v2/` docs URL for cloud products (currently set as the "default" API, `[product]/api`)
- Refactors `generate-api-docs` to remove specific conditions and use the new directory structure.
- Redocly update fixes some theme formatting issues

- Tested with the following URLs:
   
   - http://localhost:1313/influxdb/v2/api/
   - http://localhost:1313/influxdb/v2/api/v1-compatibility/
   - http://localhost:1313/influxdb/cloud/api/
   - http://localhost:1313/influxdb/cloud/api/v2/
   - http://localhost:1313/influxdb/cloud-dedicated/api/
   - http://localhost:1313/influxdb/cloud-dedicated/api/v2/
   - http://localhost:1313/influxdb/cloud/api/v1-compatibility/
   - http://localhost:1313/influxdb/cloud-dedicated/api/v1-compatibility/
   - http://localhost:1313/influxdb/cloud-serverless/
   - http://localhost:1313/influxdb/cloud-serverless/api/v2/
   - http://localhost:1313/influxdb/cloud-serverless/api/v1-compatibility/
   - http://localhost:1313/influxdb/clustered/api/
   - http://localhost:1313/influxdb/clustered/api/v2/
   - http://localhost:1313/influxdb/clustered/api/v1-compatibility/

image

image

### PR DESCRIPTION
**Pull request by jstirnaman**
_(I expect to do future work to streamline the docs generation and simplify the config.)_

This PR closes #5397 with the following changes:

- Adds an API reference doc for the  HTTP management API openapi spec (https://github.com/influxdata/granite/blob/main/openapi.yaml) (Cloud Dedicated only), hidden from sitemap and menus.
- Adds Management API examples from https://raw.githubusercontent.com/influxdata/granite/aea169033af0d7f7d6124c1a3cab61c085038bd2/scripts/test_http_api_v0_endpoints.sh
- **Doesn't** fetch changes for any specs except for the Management spec.
- Adds summaries and fixes descriptions in spec overrides.
- Removes versions in spec overrides.
- Adds configuration for generating API reference docs, Hugo frontmatter, setting the default API (for the [product]/api/ URL)
- Improves Hugo frontmatter descriptions and reuses titles and summaries from specs.
- Restructures `/api-docs` to better allow for multiple APIs (and customizations) per product
- Adds `[product]/api/v2/` docs URL for cloud products (currently set as the "default" API, `[product]/api`)
- Refactors `generate-api-docs` to remove specific conditions and use the new directory structure.
- Redocly update fixes some theme formatting issues

- Tested with the following URLs:
   
   - http://localhost:1313/influxdb/v2/api/
   - http://localhost:1313/influxdb/v2/api/v1-compatibility/
   - http://localhost:1313/influxdb/cloud/api/
   - http://localhost:1313/influxdb/cloud/api/v2/
   - http://localhost:1313/influxdb/cloud-dedicated/api/
   - http://localhost:1313/influxdb/cloud-dedicated/api/management/
   - http://localhost:1313/influxdb/cloud-dedicated/api/v2/
   - http://localhost:1313/influxdb/cloud/api/v1-compatibility/
   - http://localhost:1313/influxdb/cloud-dedicated/api/v1-compatibility/
   - http://localhost:1313/influxdb/cloud-serverless/
   - http://localhost:1313/influxdb/cloud-serverless/api/v2/
   - http://localhost:1313/influxdb/cloud-serverless/api/v1-compatibility/
   - http://localhost:1313/influxdb/clustered/api/
   - http://localhost:1313/influxdb/clustered/api/v2/
   - http://localhost:1313/influxdb/clustered/api/v1-compatibility/

image

image

### CODE REVIEW

**Comment by jstirnaman** [comment]
Ah, wait, that's not supposed to show in the menu :/

**Comment by jstirnaman** [comment]
> Ah, wait, that's not supposed to show in the menu :/

Fixed.

File: api-docs/cloud-dedicated/v2/content/info.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+title: InfluxDB v2 HTTP API for Cloud Dedicated
```
Comment (sanderson) [inline api-docs/cloud-dedicated/v2/content/info.yml:1]:
```suggestion
title: InfluxDB v2 HTTP API for InfluxDB Cloud Dedicated
```

**Review by sanderson — APPROVED** [review]
This looks really good. Thanks for doing this. I only had one suggestion.

**Pull request merged**